src/System.Private.ServiceModel/tools/setupfiles/sync-pr.cmd

```diff
@@ -13,36 +13,65 @@ echo.
     echo   [%~n0] A URL must be specified for the pull request synchronization URL
-    echo     Usage:  %~n0 [sync url] [optional pr-number]
-    echo   sync-url  - URL on remote server for PR synchronization
-    echo   pr-number - PR number to sync to
+    echo     Usage:  %~n0 [branch/pr] [sync-url] [optional branch-name / pr-number]
+    echo   
+    echo   branch/pr   - Sync to branch or PR [choose one]
+    echo   sync-url    - URL on remote server for PR synchronization
+    echo   branch-name - branch name to sync to
+    echo   pr-number   - PR number to sync to
     echo.
-    echo   Example:  %~n0 http://wcfci-sync-server/repo/sync.ashx 
-    echo   Example:  %~n0 http://wcfci-sync-server/repo/sync.ashx 404
+    echo   Example:  %~n0 branch http://wcfci-sync-server/PRService1/pr.ashx 
+    echo   Example:  %~n0 branch http://wcfci-sync-server/PRService1/pr.ashx master
+    echo   Example:  %~n0 pr http://wcfci-sync-server/PRService1/pr.ashx 
+    echo   Example:  %~n0 pr http://wcfci-sync-server/PRService1/pr.ashx 404
     set __EXITCODE=1
     goto done
 )
 
-set __SYNC_HOST_URL=%1
+set __OPERATION_MODE=%1
 
-if "%2"=="" (
-    if '%ghprbPullId%'=='' (
-        echo   [%~n0] This script should only be called only from the context of a GitHub Pull Request
-        echo   [%~n0] 'ghprbPullId' environment variable not set
-        set __EXITCODE=1
-        goto done
-    ) else ( 
-        set __PR_ID=%ghprbPullId%
+if not "%__OPERATION_MODE%"=="branch" if not "%__OPERATION_MODE%"=="pr" (
+    echo   Invalid operation mode specified 
+    goto done
+)
+
+set __SYNC_HOST_URL=%2
+
+if "%3"=="" (
+    if '%__OPERATION_MODE%'=='branch' (
+        if '%GIT_BRANCH%'=='' (
+            echo   [%~n0] This script should only be called only from the context of a GitHub Pull Request
+            echo   [%~n0] 'GIT_BRANCH' environment variable not set and no branch specified
+            set __EXITCODE=1
+            goto done
+        ) else ( 
+            set __BRANCH_OR_PR_ID=%GIT_BRANCH%
+        )
+    ) else if '%__OPERATION_MODE%'=='pr' (
+        if '%ghprbPullId%'=='' (
+            echo   [%~n0] This script should only be called only from the context of a GitHub Pull Request
+            echo   [%~n0] 'ghprbPullId' environment variable not set and no PR ID specified
+            set __EXITCODE=1
+            goto done
+        ) else ( 
+            set __BRANCH_OR_PR_ID=%ghprbPullId%
+        )
     )
 ) else (
-    set __PR_ID=%2
+    set __BRANCH_OR_PR_ID=%3
     echo   [%~n0] WARNING: This script should usually only be called only from the context of a GitHub Pull Request
-    echo   [%~n0] PR ID overridden: '%2'
+    echo   [%~n0] Branch or PR ID overridden: '%3'
 ) 
 
-set __REQUEST_URL=%__SYNC_HOST_URL%?pr=%__PR_ID%
+if "%__OPERATION_MODE%"=="branch" (
+    set __REQUEST_URL=%__SYNC_HOST_URL%?branch=%__BRANCH_OR_PR_ID%
+) 
+if "%__OPERATION_MODE%"=="pr" (
+    set __REQUEST_URL=%__SYNC_HOST_URL%?pr=%__BRANCH_OR_PR_ID%
+)
```

Comment:
NIT: Since you've validated the value of `__OPERATION_MODE`, you can just do
`set __REQUEST_URL=%__SYNC_HOST_URL%?%__OPERATION_MODE%=%__BRANCH_OR_PR_ID%`